doc/Language/subscripts.pod6

```diff
@@ -253,68 +251,74 @@ dimensions>) the subscript is preserved across the slice operation
-problems than for actual use.  Since some lazy sequences cache their
-results, every time they are used in a truncation, they accumulate one
-more known element.  Things like this should probably be avoided rather
-than used for effect:
+    =begin code
+    my  @letters;
+    say @letters; # OUTPUT: «[]␤»
 
-    =begin code :skip-test
-    my @a = 2, 3 ... *;
-    say flat @letters[0, 7, @a]; # OUTPUT: «(a (Any) c d e f)␤»
-    say flat @letters[0, 7, @a]; # OUTPUT: «(a (Any) c d e f (Any))␤»
+    @letters[^10] = 'a'..'z';
+    say @letters; # OUTPUT: «[a b c d e f g h i j]␤»
     =end code
 
-The runaway protection is not perfect.  The indices are eagerly evaluated,
-with the only stop condition being truncation.  This is to provide
-mostly consistent results when there is self-reference/mutation inside
-the indices.  As such, the following will most likely hang until all
-memory has been consumed:
+If you want the resulting slice to only include defined elements, you can
+silently skip the non-defined elements using the L<#:v> adverb.
 
-    =begin code :preamble<my @letters;>
-    @letters[0 xx *];
+    =begin code
+    my  @letters = <a b c d e f>;
+    say @letters[3..7]:v;  # OUTPUT: «(d e f)␤»
     =end code
 
-So, to safely use lazy indices, they should be one-shot things which
-are guaranteed to overrun the array.  The following alternate formulation
-will produce a fully lazy result (but will not truncate):
+The behavior when indexing a collection via lazy subscripts is different than
+when indexing with their eager counterparts. When accessing via a lazy
+subscript, the resulting slice will be truncated.
 
-    =begin code :preamble<my @letters;>
-    my $a = (0 xx *).map({ @letters[$_] }); # "a", "a", "a" ... forever
+    =begin code :preamble<my @letters = <a b c d e f>;>
+    say @letters[lazy 3..7]; # OUTPUT: «(d e f)␤»
+    say @letters[     3..*]; # OUTPUT: «(d e f)␤»
     =end code
 
-If you I<don't> want to specify your slice as a range/sequence but still want
-to silently skip nonexistent elements, you can use the L<#:v> adverb.
+This behavior exists as a precaution to prevent runaway generation of massive,
+potentially infinite C<Lists> and the out-of-memory issues that occur as a
+result.
+
+Behind the scenes, the implementation is first eagerly accessing the collection
+via subscripts that are already L<reified|/language/glossary#index-entry-Reify>.
+After reaching an index of the lazy subscript-producer that has not yet been
+reified, the indices will be produced one-after-another up until the point that
+the subscript no longer yields a defined value from the collection. In other
+words, once an undefined value in the collection is encountered, the process of
+subscript reification is halted and the resulting slice is returned.
+
+If you are subscripting via a lazy C<List>, be aware that each subsequent use
+reifies and caches an additional element in the lazy C<List>, introducing
```

Comment:
*[feedback]*

This problem too feels implementation-specific IMO and probably shouldn't be listed here. I opened a ticket for this problem in Rakudo: https://github.com/rakudo/rakudo/issues/1320 

It likely won't get fixed and in such a case a better place for this might be in [Reification](https://docs.perl6.org/language/glossary#index-entry-Reify) docs or in [Traps](https://docs.perl6.org/language/traps).